aiida_common_workflows/workflows/relax/generator.py

```diff
@@ -68,28 +85,49 @@ def get_builder(
         threshold_forces=None,
         threshold_stress=None,
         previous_workchain=None,
+        is_insulator=False,
+        spin=SpinType.NONE,
+        initial_magnetization='auto',
         **kwargs
     ):
         """Return a process builder for the corresponding workchain class with inputs set according to the protocol.
 
-        :param structure: the structure to be relaxed
-        :param calc_engines: ...
-        :param protocol: the protocol to use when determining the workchain inputs
-        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`
+        :param structure: the structure to be relaxed.
+        :param calc_engines: a dictionary containing the computational resources for the relaxation.
+        :param protocol: the protocol to use when determining the workchain inputs.
+        :param relaxation_type: the type of relaxation to perform, instance of `RelaxType`.
         :param threshold_forces: target threshold for the forces in eV/Å.
         :param threshold_stress: target threshold for the stress in eV/Å^3.
         :param previous_workchain: a <Code>RelaxWorkChain node.
+        :param is_insulator: a bool to activate insulator options (by default metal calcs are assumed.
+        :param spin: the spin state of the calculation, instance of `RelaxType`.
+        :param initial_magnetization: a list with the spin polarization for each site.
```

Comment:
```suggestion
        :param initial_magnetization: a list with the initial spin polarization for each site. If not defined, the builder will automatically define the initial magnetization if and only if `spin != SpinType.NONE`.
```
We should also add the units here. I think you wrote elsewhere that it is the difference in electrons in up channel vs down channel? So it is a float in units of electrons? Does this also make sense for noncollinear calculations? Do we need more information there?

Reply:
Ok units. We decided to deal with non-collinear cases in the future, not now. For sure we need angles.

Reply:
If we will need angles in the future, we should already think how they will have to be passed in the future. I have never used noncollinearity myself so I am not sure how this is normally defined, so cannot necessarily suggest anything. Would this require a different format for `initial_magnetization`? Or is an additional argument required? Is there any interface that we can choose now that will make it easy to extend in the future?

Reply:
I think the extension will simply be that the list will not accept floats or integers, but a dictionary or a list containing more info per site. The interface will remain the same and the current implementation is good. We skip your other suggestion of checking if the list contain floats or integer and we then just have to decide what to accept.

Reply:
Perhaps when a list of list is passed we can interpret it as  vectors on each site, and when a list of dict is passed the keys of the dictionary will be used, such as `angle`, `magnitude` etc.

Reply:
This is something we have to agree on, but the implementation will not change, except maybe adding few checks. I will write an issue.